src/hooks/use-sync-states-progress-info.ts

```diff
@@ -130,12 +137,59 @@ export function useSyncStatesProgressInfo() {
 		[]
 	);
 
+	const getBackupStatusWithProgress = useCallback(
+		(
+			hasBackupCompleted: boolean,
+			pullStatesProgressInfo: PullStateProgressInfoValues,
+			response: SyncBackupResponse
+		) => {
+			const frontendStatus = hasBackupCompleted
+				? pullStatesProgressInfo.downloading.key
+				: response.status;
+			let newProgressInfo: PullStateProgressInfo | null = null;
+			if ( response.status === 'in-progress' ) {
+				newProgressInfo = pullStatesProgressInfo[ frontendStatus ];
+				newProgressInfo.progress =
+					IN_PROGRESS_INITIAL_VALUE + IN_PROGRESS_TO_DOWNLOADING_STEP * ( response.percent / 100 );
+			}
+			const statusWithProgress =
+				newProgressInfo ||
+				pullStatesProgressInfo[ frontendStatus ] ||
+				pullStatesProgressInfo.failed;
```

Comment:
```suggestion
			const statusWithProgress =
				newProgressInfo ||
				pullStatesProgressInfo[ frontendStatus ];
```
Typescript already guarantees that `frontendStatus` is a key of `pullStatesProgressInfo`, so we don't need to default to `pullStatesProgressInfo.failed`.

Reply:
Good point, I've accepted the suggestion